include/nccl_ofi_math.h

```diff
@@ -39,7 +39,11 @@ extern "C" {
  * @param	a
  *		Must be a power of two
  */
+#ifndef __cplusplus
 #define NCCL_OFI_IS_ALIGNED(x, a) (((x) & ((typeof(x))(a) - 1)) == 0)
```

Comment:
typeof() should still work in C++ on GCC/Clang, right?  Can't we just make this change after the C++ switch and not have the ugly dual path?

Reply:
typeof does not work with newer gcc in C++ mode.

We actually already have something in-tree that's having to work around this: https://github.com/aws/aws-ofi-nccl/blob/master/include/contrib/uthash.h#L43-L61

But I don't want to take a dependency on that header, because I'd like to not use uthash/utlist long term. I will change it to `__typeof__`, which I think should work in all cases per the discussion on this issue: https://github.com/sheredom/utest.h/issues/19

Reply:
Nah; if typeof() is broken in recent versions of GCC, I'm ok with this patch.